1_wqpdata/src/wqp_pull.R

```diff
@@ -80,20 +81,101 @@ partition_inventory <- function(inventory_ind, wqp_pull, wqp_state_codes, wqp_co
     dplyr::left_join(parameter_codes, by='Constituent') %>%
     dplyr::mutate(SiteType=list(site_types))
 
-  return(pull_tasks_df)
+  # write the data file and the indicator file
+  data_file <- as_data_file(ind_file)
+  saveRDS(pull_tasks_df, file=data_file)
+  gd_put(ind_file, data_file) # sc_indicate(ind_file, data_file=data_file)
+
+  invisible()
+}
+
+plan_wqp_pull_per_constituent <- function(constituents, folders, folders_item) {
+
+  partitions <- scipiper::create_task_step(
+    step_name = 'partitions',
+    target_name = function(task_name, step_name, ...) {
+      sprintf('partitions_%s', task_name)
+    },
+    command = function(task_name, ...) {
+      sprintf("readRDS('%s')",
+              file.path(folders$tmp, sprintf("partition_%s.rds", task_name)))
+    }
+  )
+
+  make_plan <- scipiper::create_task_step(
+    step_name = 'make_plan',
+    target_name = function(task_name, step_name, ...) {
+      sprintf('wqp_pull_plan_%s', task_name)
+    },
+    command = function(task_name, ...) {
+      sprintf("plan_wqp_pull(%s, I('%s'), %s)",
+              sprintf('partitions_%s', task_name),
+              task_name,
+              folders_item)
+    }
+  )
+
+  create_plan <-  scipiper::create_task_step(
+    step_name = 'create_plan',
+    target_name = function(task_name, step_name, ...) {
+      sprintf('tasks_1_wqp_%s.yml', task_name)
+    },
+    command = function(task_name, ...) {
+      sprintf(
+        paste(
+          "create_task_makefile(",
+          "makefile=target_name,",
+          "task_plan=%s,",
+          "include=I(c('1_wqpdata.yml',target_name)),",
```

Comment:
I think this might be where you're getting the duplicate targets from, including `target_name` as an `include`. In this specific case, the inclusion even seems to be recursive - this remake file will be including itself! So taking that out should help.

If that's not enough, then (1) sorry, i don't know what the problem is yet, and so (2) as you're continuing to debug, keep in mind the more general rule that `include` goes all the way up the chain - if remake file C includes only remake file B, but remake file B includes remake file A, then remake file C is aware of all targets in remake file A despite not having included it directly. So you'd get a duplicate target error if you tried to directly include both B and A from C.